lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2208,17 +2207,9 @@ def draw_text_woven(chunks):
                                 font.fname, glyph_name)
                             self.file.output(Name(name), Op.use_xobject)
                             self.file.output(Op.grestore)
-
-                        # Move the pointer based on the character width
-                        # and kerning
-                        glyph = font.load_char(ccode, flags=LOAD_NO_HINTING)
-                        if lastgind is not None:
-                            kern = font.get_kerning(
-                                lastgind, gind, KERNING_UNFITTED)
-                        else:
-                            kern = 0
-                        lastgind = gind
-                        newx += kern / 64 + glyph.linearHoriAdvance / 65536
+                        newx += (
+                            font.load_glyph(char_idx, flags=LOAD_NO_HINTING)
+                            .linearHoriAdvance / 65536)
```

Comment:
It's admittedly a bit subtle: this has no effect (because newx is going to be overwritten anyways) *except* at the last iteration of the loop, where this ensures that newx is then the position including the advance of the last character (even when first running with mode=1) so that later, when running with mode=2, newx starts at a correct value.

I pushed a second commit unraveling a bit the flow of draw_text_woven (the function here); I was planning to make it a separate PR but I guess it's complicated enough that simplifying the flow may make this easier to review...

Reply:
I did somewhat see how it was used here before, but it's definitely much clearer now.